connaisseur/flask_application.py

```diff
@@ -24,6 +25,15 @@
 CONFIG = Config()
 DETECTION_MODE = os.environ.get("DETECTION_MODE", "0") == "1"
 
+metrics = PrometheusMetrics(
+    APP,
+    defaults_prefix=NO_PREFIX,
+    buckets=(0.1, 0.25, 0.5, 0.75, 1.0, 2.5, float("inf")),
```

Comment:
I am not 100% certain about the buckets, but if I see that correctly, it will allow to get a request duration histogram, right?
Essentially, it should sort requests into buckets based on their duration, right?

If that is so, I would propose to adjust the default buckets. The maximum response duration of an [admission webhook is limited to 30s](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#timeouts). Thus, the following may make more sense to cover the range:

```suggestion
    buckets=(0.1, 0.25, 0.5, 1.0, 2.5, 5.0, 10.0, 15.0, 20.0, float("inf")),
```
thus, we'd be covering the whole range and see how close we get to timeout

Reply:
> Essentially, it should sort requests into buckets based on their duration, right?
Yep! That's exactly that.

Usually, I suggest buckets created based on the general assumption of latency. In this case, anything taking between 0 <= x<= 2.5 seconds is considered "acceptable" and > 2.5s would be "too slow". That's a personal preference, But I have no issues adding more buckets.

Reply:
I see your point and agree for a common request, but in this case Connaisseur has to perform quite some verification and both `notary` as well as `cosign` take some time to verify the signatures. From personal experience, that can easily take a few seconds.
However, I might be wrong here as latency might just mean the time it takes for the request to reach the hook and I am probably talking about processing time as in `latency + processing time = response time` as discussed [here](https://stackoverflow.com/questions/58082389/what-is-the-difference-between-latency-and-response-time).

Reply:
@xopham, I agree with your points. I also added the `0.75` to the list you suggested 👍